src/architecture/serialization.md

```diff
@@ -1,533 +1,201 @@
-let expected_buffer = Vec::<u8>::from_hex(&expected_buffer_str)
-    .unwrap();
-assert_eq!(my_wallet.into_bytes(), expected_buffer);
 ```
 
-Serialized representation of `my_wallet`:
-
-| Position | Stored data  | Hexadecimal form | Comment |
-|:--------|:------:|:---------------------|:--------------------------------------------------|
-0..32  |       | `99 ac e6 c7 21 db 29 3b 0e d5 b4 87 e6 d6 11 1f 22 a8 c5 5d 2a 1b 76 06 b6 fa 6e 6c 29 67 1a a1` | Public key |
-32..36  | 48    | `30 00 00 00`            | A little endian segment pointer that refers to the string position in the serialization buffer |
-36..40  | 6     | `06 00 00 00`            | A little endian segment size |
-40..48 | 1234   | `d2 04 00 00 00 00 00 00` | A number in little endian format |
-48..54 | Andrew | `41 6e 64 72 65 77`       | UTF-8 string converted into a byte array |
-
-[transactions_macro]: https://docs.rs/exonum/*/exonum/macro.transactions.html
-[encoding_struct_macro]: https://docs.rs/exonum/*/exonum/macro.encoding_struct.html
-[zero_copy]: https://en.wikipedia.org/wiki/Zero-copy
-[asn_der]: https://en.wikipedia.org/wiki/X.690#DER_encoding
-[wiki_protobuf]: https://en.wikipedia.org/wiki/Protocol_Buffers
-[cap_n_proto]: https://capnproto.org/
-[cap_n_proto_canonicalization]: https://capnproto.org/encoding.html#canonicalization
-[sbe]: https://github.com/real-logic/simple-binary-encoding
-[wiki_flatbuf]: https://en.wikipedia.org/wiki/FlatBuffers
-[rust_primitive_types]: https://doc.rust-lang.org/book/first-edition/primitive-types.html
-[utf8]: https://en.wikipedia.org/wiki/UTF-8
-[rust_structs]: https://doc.rust-lang.org/book/first-edition/structs.html
-[rust_enums]: https://doc.rust-lang.org/book/first-edition/enums.html
-[cryptocurrency]: https://github.com/exonum/exonum/blob/master/examples/cryptocurrency
-[rust-slice]: https://doc.rust-lang.org/book/first-edition/primitive-types.html#slicing-syntax
-[rust]: http://rust-lang.org/
-[cargo_features]: https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section
-[subnormal_fp]: https://en.wikipedia.org/wiki/Denormal_number
+Note that it is required to indicate the protobuf structure to which the
+current structure refers, in the case above `proto::TransactionRequest`.
+
+[protobuf]: https://developers.google.com/protocol-buffers/docs/overview
+[proto-files]: https://github.com/exonum/exonum/tree/master/exonum/src/proto/schema/exonum
+[language]: https://developers.google.com/protocol-buffers/docs/reference/proto3-spec
+[cryptocurrency]: https://github.com/exonum/exonum/blob/master/examples/cryptocurrency/src/proto/cryptocurrency.proto
+[convert]: https://github.com/exonum/exonum/blob/master/exonum/src/proto/mod.rs
+[module]:https://github.com/exonum/exonum/tree/master/examples/cryptocurrency/src/proto
+[derive]: https://github.com/exonum/exonum/tree/master/components/derive
+[build]: https://github.com/exonum/exonum/tree/master/components/build/
```

Comment:
Nit: Why not provide links to `crates.io` or `docs.rs` for `-build` and `-derive` crates?